modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java

```diff
@@ -1901,22 +1901,26 @@ private void _notifySubscribers(
 		BlogsGroupServiceSettings blogsGroupServiceSettings =
 			BlogsGroupServiceSettings.getInstance(entry.getGroupId());
 
+		Serializable sendEmailEntryUpdatedAttributeValue =
+			serviceContext.getAttribute("sendEmailEntryUpdated");
+
 		boolean sendEmailEntryUpdated = GetterUtil.getBoolean(
-			serviceContext.getAttribute("sendEmailEntryUpdated"));
+			sendEmailEntryUpdatedAttributeValue);
 
-		Map<String, Serializable> attributes = serviceContext.getAttributes();
+		boolean updateTriggeredByScheduler = false;
 
-		boolean updateTriggeredByScheduler = !attributes.containsKey(
-			"sendEmailEntryUpdated");
+		if (sendEmailEntryUpdatedAttributeValue == null) {
+			updateTriggeredByScheduler = true;
+		}
 
 		if (serviceContext.isCommandAdd() &&
 			blogsGroupServiceSettings.isEmailEntryAddedEnabled()) {
 		}
-		else if ((sendEmailEntryUpdated ||
+		else if (serviceContext.isCommandUpdate() &&
+				 blogsGroupServiceSettings.isEmailEntryUpdatedEnabled() &&
+				 (sendEmailEntryUpdated ||
 				  (updateTriggeredByScheduler &&
-				   serviceContext.isCommandUpdate())) &&
-				 serviceContext.isCommandUpdate() &&
-				 blogsGroupServiceSettings.isEmailEntryUpdatedEnabled()) {
+				   serviceContext.isCommandUpdate()))) {
```

Comment:
On this point the `serviceContext.isCommandUpdate()` is always true. So maybe is not needed.
but if you want to assure it in case it changes, is ok

and in other hand `updateTriggeredByScheduler` there is not need to be created, null check here (`sendEmailEntryUpdatedAttributeValue == null`) will do the same.

but this changes are suggestions only.

LGTM

Reply:
@laszlopap Why don't use an explicit serviceContext attribute to indicate that we're invoking this from a scheduler? It is a bit opaque (at least to me) whether the condition is always true when invoked from a scheduler (and only true in that case).
I'd prefer something similar to this:
```
Boolean updateTriggeredByScheduler = (Boolean)serviceContext.getAttribute(FooConstants.SCHEDULER_INVOCATION);

...

if (updateTriggeredByScheduler != null && updateTriggeredByScheduler) {
    // your logic here.
} ...
```
You'd need to rename the variable to match the constant name too.

Reply:
Okay @adolfopa ! Thanks for your suggestion. Until I finish the implementation of this new approach, can I deliver an SDH to the customer?

Reply:
Unless you can 100% guarantee that the original updateTrigger condition runs only when invoked from an scheduler and this is a serious issue (it doesn't look like a critical thing to me, but you may have additional information), I'd prefer to send the customer the final solution.